pkg/cluster/mpi/configmap.go

```diff
@@ -14,12 +15,47 @@ import (
 	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/core"
 )
 
-var (
-	sshdConfigTmpl = `
-Port %d
-AuthorizedKeysFile %s
-`
-)
+const launchScriptTemplate = `#!/bin/bash
+
+set -o nounset 
+set -o errexit
+
+SSH_PORT=%[1]d
+DOMINO_UID=%[2]d
+DOMINO_GID=%[3]d
+DOMINO_USER=%[4]s
+DOMINO_GROUP=%[5]s
+AUTHORIZED_KEYS_PATH=%[6]s
+
+if ! id $DOMINO_UID >/dev/null 2>&1; then
+    groupadd -g $DOMINO_GID $DOMINO_GROUP
+    useradd -u $DOMINO_UID -g $DOMINO_GID -mN -s /bin/bash $DOMINO_USER
```

Comment:
If useradd or su etc are missing, what will happen? I believe in other failure modes, we end up with a cluster stuck in a pending state with workers in a crash loop backoff. Same here?

Reply:
Correct. The script will fail right away, and Kubernetes will try to recreate pods. There is an outstanding bug to handle this in the controller properly.